ctrd/client_opts.go

```diff
@@ -59,3 +65,43 @@ func WithDefaultNamespace(ns string) ClientOpt {
 		return nil
 	}
 }
+
+// WithInsecureRegistries sets the insecure registries to allow http request
+// and skip secure verify.
+func WithInsecureRegistries(endpoints []string) ClientOpt {
+	return func(c *clientOpts) error {
+		registries := make([]string, 0, len(endpoints))
+
+		for _, r := range endpoints {
+			if strings.Contains(strings.ToLower(r), "://") {
+				return fmt.Errorf("insecure registry %s should not contain any '://'", r)
+			}
+
+			if err := validateHostPort(r); err != nil {
+				return err
+			}
+			registries = append(registries, r)
+		}
+		c.insecureRegistries = registries
+		return nil
+	}
+}
+
+func validateHostPort(s string) error {
+	_, port, err := net.SplitHostPort(s)
+	if err != nil {
+		port = ""
```

Comment:
why not return err？

Reply:
maybe it is domain name without any port, such as 80